x/model/types/messages_model.go

```diff
@@ -203,7 +205,7 @@ func (msg *MsgUpdateModel) ValidateBasic() error {
 		return err
 	}
 
-	if msg.EnhancedSetupFlowOptions == 0 {
+	if msg.EnhancedSetupFlowOptions&1 == 1 {
```

Comment:
Do we have a logic and tests that EnhancedSetupFlow fields must be present IF AND ONLY IF EnhancedSetupFlowOptions&1 == 1 when Updating a Model?

Reply:
Added tests